package/MDAnalysis/lib/log.py

```diff
@@ -296,11 +320,12 @@ def echo(self, step, **kwargs):
             return
         self.update(step, **kwargs)
         format = self.format
+        newline = not self.dynamic
         if self.step == self.numsteps:
-            if self.last_newline:
-                format = self.format[:-1] + self.last_newline
+            newline = True
         elif self.numouts % self.interval == 0:
             pass
         else:
             return
-        echo(format % vars(self))
+        echo(format.format(**vars(self)),
```

Comment:
This will break a lot of progmeter code using the old formatting style that's needs to be updated as well. It should also be mentioned in the changelog explicitly